hudi-common/src/main/java/org/apache/hudi/common/table/timeline/HoodieActiveTimeline.java

```diff
@@ -212,25 +224,24 @@ public void saveAsComplete(HoodieInstant instant, Option<byte[]> data) {
     log.info("Completed " + instant);
   }
 
-  public void revertToInflight(HoodieInstant instant) {
-    log.info("Reverting " + instant + " to inflight ");
-    revertStateTransition(instant, HoodieTimeline.getInflightInstant(instant));
-    log.info("Reverted " + instant + " to inflight");
-  }
-
-  public HoodieInstant revertToRequested(HoodieInstant instant) {
-    log.warn("Reverting " + instant + " to requested ");
-    HoodieInstant requestedInstant = HoodieTimeline.getRequestedInstant(instant);
-    revertStateTransition(instant, HoodieTimeline.getRequestedInstant(instant));
-    log.warn("Reverted " + instant + " to requested");
-    return requestedInstant;
+  public HoodieInstant revertToInflight(HoodieInstant instant) {
+    log.info("Reverting instant to inflight " + instant);
+    HoodieInstant inflight = HoodieTimeline.getInflightInstant(instant, metaClient.getTableType());
+    revertCompleteToInflight(instant, inflight);
```

Comment:
how do such reverts work now? we just seem to delete the complete instant and it will for e.g make it inflight again? is delete() atomic in cloud stores?

Reply:
Deletes are atomic in GCS but not so in S3. Thought about this. It shouldn't bring in new failure scenarios here.